source/extensions/filters/http/transformation/transformation_filter.cc

```diff
@@ -205,6 +209,29 @@ void TransformationFilter::addEncoderData(Buffer::Instance &data) {
   encoder_callbacks_->addEncodedData(data, false);
 }
 
+void TransformationFilter::transformOnStreamCompletion() {
+  if (on_stream_completion_transformation_ == nullptr) {
+    return;
+  }
+
+  Http::StreamFilterCallbacks *on_complete_callbacks{};
+  // Body isn't required for this transformer since it isn't included
+  // in access logs
+  Buffer::OwnedImpl emptyBody{};
+  try {
+    on_stream_completion_transformation_->transform(*response_headers_,
```

Comment:
There might be an edge case where client disconnects, and then response headers will be nullptr (because the stream was completed without a response). we should probably allow response_headers to be nullptr in this case

Reply:
I don't think I follow exactly. Should that be supported at this stage or would we expect it to be handled in the transformer? I think as is, this call will accept a response header that is a nullptr. Are you thinking we shouldn't transform if the client disconnects?

Reply:
Actually I think my assumption was wrong. I'll update the interface to accept nullptr here!

Reply:
Instead of updating the interface for `transform` to accept a nullptr, I added in a check that sets `response_headers_` to an empty header map if it's a nullptr. Is that a reasonable solution?